tests/unit/scribbles/test_transforms.py

```diff
@@ -0,0 +1,292 @@
+            "prob": three_dim_data["prob"],
+            "scribbles": three_dim_data["prob"][[1], ...] + 2,
+        },
+        {"target": three_dim_data["target"]},
+        (1, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+TEST_CASE_MAKE_ISEG_UNARY_TX = [
+    # 2D case
+    (
+        {
+            "image": "image",
+            "logits": "prob",
+            "scribbles": "scribbles",
+            "scribbles_bg_label": 2,
+            "scribbles_fg_label": 3,
+        },
+        {"image": two_dim_data["image"], "prob": two_dim_data["prob"], "scribbles": two_dim_data["prob"][[1], ...] + 2},
+        {"target": two_dim_data["prob"]},
+        (2, HEIGHT, WIDTH),
+    ),
+    # 3D case
+    (
+        {
+            "image": "image",
+            "logits": "prob",
+            "scribbles": "scribbles",
+            "scribbles_bg_label": 2,
+            "scribbles_fg_label": 3,
+        },
+        {
+            "image": three_dim_data["image"],
+            "prob": three_dim_data["prob"],
+            "scribbles": three_dim_data["prob"][[1], ...] + 2,
+        },
+        {"target": three_dim_data["prob"]},
+        (2, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+TEST_CASE_MAKE_LIKE_HIST_TX = [
+    # 2D case
+    (
+        {"image": "image", "scribbles": "scribbles", "scribbles_bg_label": 2, "scribbles_fg_label": 3},
+        {"image": two_dim_data["target"], "scribbles": two_dim_data["prob"][[1], ...] + 2},
+        {"target": two_dim_data["prob"]},
+        (2, HEIGHT, WIDTH),
+    ),
+    # 3D case
+    (
+        {"image": "image", "scribbles": "scribbles", "scribbles_bg_label": 2, "scribbles_fg_label": 3},
+        {"image": three_dim_data["target"], "scribbles": three_dim_data["prob"][[1], ...] + 2},
+        {"target": three_dim_data["prob"]},
+        (2, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+
+TEST_CASE_ADD_BG_ROI = [
+    (
+        {"scribbles": "scribbles", "roi_key": "roi", "scribbles_bg_label": 2, "scribbles_fg_label": 3},
+        {
+            "scribbles": np.zeros((1, NUM_SLICES, HEIGHT, WIDTH), dtype=np.float32),
+            "roi": [
+                NUM_SLICES // 2 - 4,
+                NUM_SLICES // 2 + 4,
+                HEIGHT // 2 - 8,
+                HEIGHT // 2 + 8,
+                WIDTH // 2 - 16,
+                WIDTH // 2 + 16,
+            ],
+        },
+        (1, NUM_SLICES, HEIGHT, WIDTH),
+    ),
+]
+
+
+class TestScribblesTransforms(unittest.TestCase):
+    @parameterized.expand(TEST_CASE_ADD_BG_ROI)
+    def test_add_bg_roi_transform(self, input_param, test_input, expected_shape):
+        result = AddBackgroundScribblesFromROId(**input_param)(test_input)
+        mask = result["scribbles"].astype(bool)
+        mask[
+            :,
+            test_input["roi"][0] : test_input["roi"][1],
+            test_input["roi"][2] : test_input["roi"][3],
+            test_input["roi"][4] : test_input["roi"][5],
+        ] = True
+        target = result["scribbles"]
```

Comment:
`target` should be something predetermined before the test is run and/or independent from the output of `AddBackgroundScribblesFromROId` so you can compare the `result["scribbles"]` against, not something derived from the result of `AddBackgroundScribblesFromROId`

Reply:
e.g. `target` it should come in along with `expected_shape`

Reply:
You are right. Target needs to be built directly from roi input and then compared against output. I will correct this.

Reply:
Fixed this with using original input and making the output target on the fly.